core/src/main/java/org/elasticsearch/cluster/node/DiscoveryNode.java

```diff
@@ -312,25 +312,30 @@ public int hashCode() {
         return nodeId.hashCode();
     }
 
+    private String toString;
```

Comment:
@bleskes We use `DiscoveryNode#toString` all over the place in `ZenDiscovery` and related classes when pinging and handling cluster state processing. Just crack open `ZenDiscovery` and look at all the places where an instance of `DiscoveryNode` is concatenated with a `String` (e.g., in places where cluster state update tasks are submitted).

Here's an example from [`PublishClusterStateAction`](https://github.com/elastic/elasticsearch/blob/b86d0983bb5bbb2e08cda05fde263a281c4a0d25/core/src/main/java/org/elasticsearch/discovery/zen/publish/PublishClusterStateAction.java#L430):

```java
            newPendingClusterStatelistener.onNewClusterState("master " + state.nodes().getMasterNode() + " committed version [" + state.version() + "]");
```

Now you can see how I arrived at this *today* specifically. 😄